Compilation failure on branch-4.17: ConcurrentLongHashMap uses Java `record` but the branch targets Java 8

### Description

Branch `branch-4.17` fails to compile after commit 0e1bddd7e5 ([Fix] ConcurrentLongHashMap throw ArrayIndexOutOfBoundsException, #4771) was backported to it.

The commit introduces a Java `record` declaration in `ConcurrentLongHashMap`:

```java
private record Table<V>(long[] keys, V[] values, int capacity) { }
```

The `record` keyword requires Java 16+, but on `branch-4.17` the root `pom.xml` sets:

```xml
<maven.compiler.source>8</maven.compiler.source>
<maven.compiler.target>8</maven.compiler.target>
```

so the file fails to parse.

### Build error

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.12.1:compile (default-compile) on project bookkeeper-server: Compilation failure: Compilation failure:
[ERROR] .../ConcurrentLongHashMap.java:[320,29] ';' expected
[ERROR] .../ConcurrentLongHashMap.java:[320,32] illegal start of type
```

### Affected branches

- `branch-4.17` — broken (Java 8 target).
- `master` — not affected (Java 17 target).

### Proposed fix

Replace the record with an equivalent `static final` nested class exposing `keys()`, `values()` and `capacity()` accessors, preserving the existing call sites.

--- a/bookkeeper-server/src/main/java/org/apache/bookkeeper/util/collections/ConcurrentLongHashMap.java
+++ b/bookkeeper-server/src/main/java/org/apache/bookkeeper/util/collections/ConcurrentLongHashMap.java
@@ -317,7 +317,29 @@ public class ConcurrentLongHashMap<V> {
     // previous design had to paper over with Math.min(keys.length, values.length).
     @SuppressWarnings("serial")
     private static final class Section<V> extends StampedLock {
-        private record Table<V>(long[] keys, V[] values, int capacity) { }
+        private static final class Table<V> {
+            private final long[] keys;
+            private final V[] values;
+            private final int capacity;
+
+            Table(long[] keys, V[] values, int capacity) {
+                this.keys = keys;
+                this.values = values;
+                this.capacity = capacity;
+            }
+
+            long[] keys() {
+                return keys;
+            }
+
+            V[] values() {
+                return values;
+            }
+
+            int capacity() {
+                return capacity;
+            }
+        }
 
         // Section is Serializable only by inheritance from StampedLock; never actually serialized.
         @SuppressFBWarnings("SE_BAD_FIELD")

